pkg/controller/webhook/cert.go

```diff
@@ -0,0 +1,126 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package webhook
+
+import (
+	cryptorand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"time"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/certificates"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WebhookCertificates holds the artifacts used by the webhook server and the webhook configuration.
+type WebhookCertificates struct {
+	caCert []byte
+
+	serverKey  []byte
+	serverCert []byte
+}
+
+func (w *Params) shouldRenewCertificates(serverCertificates *corev1.Secret, webhookConfiguration *v1beta1.ValidatingWebhookConfiguration) bool {
+	// Read the current certificate used by the server
+	serverCA := certificates.BuildCAFromSecret(*serverCertificates)
+	if serverCA == nil {
+		return true
+	}
+	if !certificates.CanReuseCA(serverCA, w.Rotation.RotateBefore) {
+		return true
+	}
+	// Read the certificate in the webhook configuration
+	for _, webhook := range webhookConfiguration.Webhooks {
+		caBytes := webhook.ClientConfig.CABundle
+		if len(caBytes) == 0 {
+			return true
+		}
+		// Parse the certificates
+		certs, err := certificates.ParsePEMCerts(caBytes)
+		if err != nil {
+			log.Error(err, "Cannot parse PEM cert from webhook configuration, will create a new one", "webhook_name", webhookConfiguration.Name)
+			return true
+		}
+		if len(certs) == 0 {
+			return true
+		}
+		for _, cert := range certs {
+			if !certificates.CertIsValid(*cert, w.Rotation.RotateBefore) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (w *Params) newCertificates() (WebhookCertificates, error) {
+	webhookCertificates := WebhookCertificates{}
+
+	// Create a new CA
```

Comment:
I'd appreciate some comments about the lifecycle of the CA. IIUC it's created but not persisted anywhere (no need), and recreated when the certificates expire or are missing?